storage/replica.go

```diff
@@ -1344,12 +1344,18 @@ func (r *Replica) maybeGossipSystemConfig() {
 	r.maybeGossipSystemConfigLocked()
 }
 
+// IsLeader returns true if the node has a leader lease on the replica.
```

Comment:
This is a subtle thing to expose: leader leases are tied into the transaction timestamp system, and accessing them outside a transaction using only the current time is only best-effort and may produce anomalies (such as multiple apparent leaders, or long periods with no leaders if the range has gone dormant). 

I don't think we should expose this to higher layers of the stack. Instead, can you make a table leader lease table similar to the descriptor lease table and manage this leadership in the database?